sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -220,3 +220,79 @@ def test_singular_values():
     rpca.fit(X_hat_rpca)
     assert_array_almost_equal(apca.singular_values_, [3.142, 2.718, 1.0], 14)
     assert_array_almost_equal(rpca.singular_values_, [3.142, 2.718, 1.0], 14)
+
+
+def test_svd_whiten():
+    tsvd = TruncatedSVD(n_components=X.shape[1] - 1, random_state=42,
+                        whiten=True, algorithm='arpack')
+    Xt = tsvd.fit_transform(X)
+    Xt_2 = tsvd.transform(X)
+    assert_allclose(Xt, Xt_2)
+    Xinv = tsvd.inverse_transform(Xt)
+    # Xdense is a sparse array in dense format (check zeros separately)
+    # The below equality is approximate since n_components < n_features
+    mask = Xdense == 0.0
+    assert_allclose(Xinv[mask], 0, atol=0.08)
+    assert_allclose(Xinv[~mask], Xdense[~mask], rtol=0.04)
+
+
+def test_svd_consistency():
+    X_c = X - X.mean(axis=0)
+    svd = TruncatedSVD(n_components=5, whiten=True, random_state=0)
+    assert_allclose(svd.fit(X_c).explained_variance_,
+                    np.ones(5))
+
+
+def test_truncated_svd_eq_pca():
+    # TruncatedSVD should be equal to PCA on centered data
+
+    X_c = X - X.mean(axis=0)
+
+    pars = dict(n_components=52, random_state=42)
+
+    for whiten in [False, True]:
+        svd = TruncatedSVD(whiten=whiten, **pars)
+        pca = PCA(whiten=whiten, **pars)
```

Comment:
params. Why 52 components, a tiny number would be preferable. I think if you use the randomized algorithm for PCA these should be closer / the same?